holopy/core/metadata.py

```diff
@@ -211,20 +217,50 @@ def update_metadata(a, medium_index=None, illum_wavelen=None, illum_polarization
 
     return b
 
+
+def get_spacing(im):
+    xspacing = np.diff(im.x)
+    yspacing = np.diff(im.y)
+    if not (np.allclose(xspacing[0], xspacing) and
+            np.allclose(yspacing[0], yspacing)):
+        msg = "array has nonuniform spacing, can't determine a single spacing"
+        raise ValueError(msg)
+    return np.array((xspacing[0], yspacing[0]))
+
+
+def get_extents(im):
+    if np.ndim(im) == 1:
+        raise ValueError("Cannot get extent for detector_points")
+    def get_extent(d):
+        if len(im[d]) < 2:
+            return 0
+        # Add one extra spacing since the xarray coords are right edge only,
+        # but we actually want right edge of first pixel to left edge of last
+        # pixel
+        return float(im[d][-1] - im[d][0] + np.diff(im[d]).mean())
```

Comment:
Yeah I agree but I don't know what is going on. The two pathways are different? For instance, ```get_extents``` gets an extent for z, but ```get_spacing``` doesn't use z. Thoughts? Save for a later "clean metadata" PR?

Reply:
The part about `np.diff(im[d]).mean()` is just getting the mean spacing between pixels along dimension d (passed in as either 'x' or 'y'), which `get_spacing` does a better job of because it confirms the spacing is actually uniform. That being said, I don't see a trivially easy way to use it here because of the way `get_extents` is written.

Reply:
As we discussed, unfortunately this stays as is until a later rewrite.